tools/nnyeah/NNyeahTests/Compiler.cs

```diff
@@ -0,0 +1,73 @@
+using System;
+using System.Text;
+using System.IO;
+
+namespace Microsoft.MaciOS.Nnyeah.Tests {
+	public class Compiler {
+		const string kCompiler = "/Library/Frameworks/Mono.framework/Versions/Current/Commands/csc";
+
+		public static string CompileText (string text, string outputFile, PlatformName platformName, bool isLibrary)
+		{
+			using var provider = new DisposableTempDirectory ();
+			var outputCSFile = Path.Combine (provider.DirectoryPath, "LibraryFile.cs");
+			using var writer = new StreamWriter (outputCSFile);
+			writer.Write (text);
+			writer.Close ();
+			return Compile (outputFile, platformName, isLibrary, provider.DirectoryPath, outputCSFile);
+		}
+
+		public static string Compile (string outputFile, PlatformName platformName, bool isLibrary, string workingDirectory, params string[] sourceFiles)
+		{
+			var compilerArgs = BuildCompilerArgs (sourceFiles, outputFile, platformName, isLibrary);
+			return ExecAndCollect.Run (kCompiler, compilerArgs, workingDirectory);
+		}
+
+		static string BuildCompilerArgs (string[] sourceFiles, string outputFile, PlatformName platformName,
+			bool isLibrary)
+		{
+			var sb = new StringBuilder ();
+
+			sb.Append ("/unsafe ").Append ("/out:").Append (outputFile);
+			sb.Append (" /nostdlib+");
+			AppendPlatformReference (sb, platformName, "mscorlib");
+			AppendPlatformReference (sb, platformName, XamarinLibName (platformName));
+			sb.Append (" /debug+ /debug:full /optimize-");
+			sb.Append (" /out:").Append (outputFile);
+			sb.Append (" /target:").Append (isLibrary ? "library" : "exe");
+
+			foreach (var file in sourceFiles) {
+				sb.Append (' ').Append (file);
+			}
+
+			return sb.ToString ();
+		}
+
+		static StringBuilder AppendPlatformReference (StringBuilder sb, PlatformName platformName, string libName)
+		{
+			return sb.Append (" /reference:").Append (PlatformLibPath (platformName, libName));
```

Comment:
We could return void since we do not do any fluent call later, but is not too bad.